server/src/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1731,6 +1731,17 @@
         Long zoneId = cmd.getZoneId();
         Long podId = cmd.getPodId();
 
+        List<Long> ids = null;
```

Comment:
Would you mind extracting lines 1734-1744 to a method? Then, you could write test cases and even create some documentation for it.